homeassistant/components/recorder/migration.py

```diff
@@ -463,12 +463,15 @@ def _apply_update(engine, session, new_version, old_version):
         # This dropped the statistics table, done again in version 18.
         pass
     elif new_version == 18:
+        # Recreate the statisticsmeta tables
```

Comment:
Hi @frenck, unfortunately this still gave some errors on my system:
```
psycopg2.errors.DependentObjectsStillExist: cannot drop table statistics_meta because other objects depend on it
DETAIL:  constraint statistics_metadata_id_fkey on table statistics depends on table statistics_meta
HINT:  Use DROP ... CASCADE to drop the dependent objects too.
```

Fixed it by dropping the tables together using drop_all:
```python
    elif new_version == 18:
        # Recreate the statistics and statisticsmeta tables
        if sqlalchemy.inspect(engine).has_table(StatisticsMeta.__tablename__):
            Base.metadata.drop_all(bind=engine, tables=[StatisticsMeta.__table__, Statistics.__table__])

        StatisticsMeta.__table__.create(engine)
        Statistics.__table__.create(engine)
```

Reply:
Nice! Thanks I was working on it on #52453